tests/deprecation.rs

```diff
@@ -10,7 +10,8 @@ fn test_regressions() {
         ("0.1.0.", VersionReq::parse("0.1.0").unwrap()),
         ("0.3.1.3", VersionReq::parse("0.3.13").unwrap()),
         ("0.2*", VersionReq::parse("0.2.*").unwrap()),
-        ("*.0", VersionReq::any()),
+        // TODO: this actually parses as '*' now, not sure if that's OK
+        // ("*.0", VersionReq::any()),
```

Comment:
So, I am actually not sure what the difference here is between "parses as *" and Any, that is, they should be the same?

Reply:
When I was comparing it on <https://semver.npmjs.com/>, they both resolve to the same set of versions. It's now not "deprecated". The question is if we care about making it deprecated (want to discourage people from using this)?

Reply:
Ah yes, the deprecation. Thanks :) It is probably fine to un-deprecate it, but I'd like to triple check against https://github.com/semver/semver/pull/584 real quick to see if it's acceptable in that grammar....

Reply:
There's also some other TODOs we need to either clean up in this PR or another one before we ship a new version of both `semver` and `semver-parser` that @mikrowstew left behind.

Reply:
I think semver/semver#584 disallows it, but fwiw, node-semver parses it as `*` as well.  If y'all move forward with doing it that way, we should update the spec to make that official.

Reply:
@isaacs that sounds good to me.